aea/mail/base.py

```diff
@@ -334,9 +378,43 @@ def __init__(
         )
         if isinstance(message, Message):
             message = self._check_consistency(message, to, sender)
+
         self._to = to
         self._sender = sender
-        self._protocol_id = protocol_id
+
+        # set protocol_id and protocol_specification_id
+        if isinstance(message, Message):
+            # protocol_id provided as an argument in priority
+            # use Message.protocol_id only if no protocol_id provided
+            protocol_id = protocol_id or Message.protocol_id
+
+        # no protocol_id provided and not a Message instance for message
+        if not protocol_id:
+            # try to get protocol_id by protocol_specification_id
+            if not protocol_specification_id:
+                raise ValueError(
+                    "protocol_id or protocol_specification_id or message as instance of Message class must be provided!"
+                )
+            protocol_id = self._get_protocol_id_by_protocol_specification_id(
+                protocol_specification_id
+            )
+
+        if not protocol_specification_id:
+            # if no protocol_spcification_id, try to resolve by protocol_id
+            protocol_specification_id = self._get_protocol_specification_id_by_the_protocol_id(
+                protocol_id
+            )
```

Comment:
This and above can be massively simplified. There are only 3 valid ways:
- message is type Message and no protocol_id or specification_id provided
- message is bytes and protocol_id provided
- message is bytes and specification_id provided
Everything else should raise!